docs/release.md

```diff
@@ -0,0 +1,73 @@
+
+The release automation will be responsible for producing manifest list container images, generating rendered Kubernetes manifests, and a GithHub release that will contain both of those attached as release artifacts.
+
+## Publishing a new release
+
+### Steps
+
+- Review the current release milestone
+- Create and push a new release tag
+- Verify the goreleaser action was successful
+- Add the release installation header
+
+#### Review the current release milestones
+
+Ensure that the current release [milestone][milestone] is accurately reflected, and all tickets are in the closed state. When a release milestone has open tickets, reach out to the [rukpak-dev][slack] channel to see which tickets can be removed from the release.
+
+#### Create and push a new release tag
+
+In order to trigger the existing release automation, you need to first create a new tag locally, and push that tag up to the upstream repository remote.
+
+- Make sure you are on `main` branch.
+- Pull the latest commits from the upstream `main` branch
+- Make a new tag that matches the version.
+- Push tag directly to this repository.
+
+**Note**: The following steps assume that the upstream rukpak repository remote is named `upstream`. Replace the v0.3.0 example tag with the correct tag when running these commands locally.
+
+```bash
+git checkout main
+git pull --rebase upstream main
+# Create an annotated tag as `git describe` only uses annotated tags by default.
+git tag -a v0.3.0 -m "v0.3.0 release"
+git push upstream v0.3.0
+```
+
+#### Verify that GoReleaser is Running
+
+Once a manual tag has been created, monitor the progress of the [release workflow action][workflow] that was triggered when a new tag has been created to ensure a successful run.
+
+Once that release workflow has run, navigate to the [quay.io/operator-framework/rukpak image repository][image] and ensure that the expected container images have been created.
+
+#### Add the release installation header
+
+Navigate to the generated [release][release] and add the "Installation header":
+
+```markdown
+## Installation
+
+```bash
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.1/cert-manager.yaml
+kubectl wait --for=condition=Available --namespace=cert-manager deployment/cert-manager-webhook --timeout=60s
+kubectl apply -f https://github.com/operator-framework/rukpak/releases/download/v0.3.0/rukpak.yaml
+kubectl wait --for=condition=Available --namespace=rukpak-system deployment/plain-provisioner --timeout=60s
+kubectl wait --for=condition=Available --namespace=rukpak-system deployment/core-webhook --timeout=60s
+kubectl wait --for=condition=Available --namespace=crdvalidator-system deployment/crd-validation-webhook --timeout=60s
+```
+
+And replace the "https://github.com/operator-framework/rukpak/releases/download/v0.3.0/rukpak.yaml" command with the correct tag endpoint.
```

Comment:
I realize this can definitely be automated as well. I'll open up a PR for this and remove this section once once it lands in the main branch.

Reply:
#343